datafusion/sql/src/expr/mod.rs

```diff
@@ -476,13 +476,6 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 planner_context,
             ),
 
-            #[cfg(not(feature = "unicode_expressions"))]
-            SQLExpr::Substring { .. } => {
-                internal_err!(
-                    "statement substring requires compilation with feature flag: unicode_expressions."
-                )
-            }
```

Comment:
I think this might still be needed?

Reply:
Indeed, I considered it as matching special to false, without carefully checking.
Fixed. Thank you @Jefffrey !